### DISCUSSION TITLE
Should we only support arm64 for embedded linux arm platforms going forward?

### DISCUSSION THREAD

**Discussion by ofTheo** [openframeworks/openFrameworks#7138] [Polls]
One of the things that has made new releases and maintenance slow and tedious is the shear number of platforms OF is supporting.  

As its been several years since our last release thought it would be good to see if we can fully transition from armv6 / armv7 to arm64 for embedded linux platforms like Raspberry Pi.  

Rpi 3 and 4 support arm64. It will be a fair amount of work to get our build setup configured for arm64, so I am wondering if we can drop the other platforms?  ie: if you want to use armv6 you have to use 0.11.2 but 0.12.0 and newer is arm64 only? 

I know @dimitre has been trying out an arm64 PR here: 
https://github.com/openframeworks/openFrameworks/pull/7069 ( also a related issue #6770 ) 

Curious if people who use embedded devices more often have any thoughts? 
@madelinegannon @patriciogonzalezvivo @roymacdonald @jvcleave

**Comment by dimitre** [comment]
100% about moving forward, supporting more recent configurations. Dropping everything that makes weight (like tvOS)
In the same train of thought I think next OF should be C++17 first, and some alternative if not possible. maybe drop boost?

**Reply by danoli3** [reply]
C++17 abi is a good idea requires all dependancies at the same abi though. Yeah basically boost is merged into c++17 anyways

**Comment by armadillu** [comment]
Yes, minimize clutter! You can always use an old release if for some reason you need arm32.

**Comment by danoli3** [comment]
arm64 for all platforms including windows  

Drop armv6 / armv7 / i386 - 100%

**Comment by jvcleave** [comment]
As for the RPI I feel the same way as @armadillu

**Comment by danoli3** [comment]
Android will be the exception, need to support armv7 for that platform for sometime, however new cmake build systems make that easier

**Comment by steeley** [comment]
Yes arm64. Have been wanting to use 64 bit OF on Pi and similar. Use OF on Linux mostly, and quite a bit on OSX. Not interested in Windows or Android.

**Comment by Autr** [comment]
ARM64, since Pi Zero (2) is now also that. Perhaps also some sort of [Firmata](https://github.com/firmata/ConfigurableFirmata) example for the [Pico](https://github.com/firmata/ConfigurableFirmata/pull/117).

I would like to help out - is there an effort underway, or yet to be decided?

**Reply by GitBruno** [reply]
Pi Zero 2 runs way better on 32 bit.

**Reply by danoli3** [reply]
more likely a bug with gcc versions / target / package optimisation flags or bloat, if memory is 64 bit on 64 bit it should not be slower 

Can you confirm https://qengineering.eu/install-64-os-on-raspberry-pi-zero-2.html 64bit slice was installed and running and what FPS difference we talking 

I have a Pi 2W I'll test it out soon myself with profiler

**Reply by GitBruno** [reply]
Awesome! Please report back the numbers. I noticed the whole OS running better on 32bit. Considering the limited amount of memory on the Pi Zero this makes sense. In 32 bit we simply have more memory to play with, which those boards desperately needs.

**Reply by n1ckfg** [reply]
Available RAM on a Pi 3B+ with 1GB RAM, boot to CLI: 
Buster 32-bit, 656MB (+ fixed 256MB preallocated to GPU)
Bookworm 64-bit, 622MB (before GPU allocation)

**Reply by n1ckfg** [reply]
imo it's clear that Pis 0 through 3 work better on 32-bit; also the older OSes have broader support for their special OMX and MIPI hardware, where a lot of the older models' remaining value comes from. On the other hand this may be more of a documentation issue, recommending which OS and oF versions are best for older hardware, while development efforts go to 64-bit on newer Pis.

**Comment by n1ckfg** [comment]
I'd be interested in hearing other folks' experiences, but I find each oF major version on Raspberry Pi has always required a corresponding new RPi OS version, and not all Pis can run all OS versions, so this doesn't break the pattern. imo the only catch is we need to warn people with RPi 2s specifically to check their CPU--that one was sold in _both_ 32-bit and 64-bit versions.

**Reply by ofTheo** [reply]
Thanks @n1ckfg 
Btw there is also some discussion and coordination happening on the [raspberry-pi channel](https://join.slack.com/t/openframeworks/shared_invite/zt-1r2brqms0-dZMMFZgZhFTgomjJ0vlCjA) on Slack

**Reply by cpietsch** [reply]
Hey, I am looking into this currently and want to join the slack channel. Can you post an invite for the slack ?

**Reply by n1ckfg** [reply]
Sent an admin request for invite

**Comment by summerlotus513** [comment]
I am also interested in this issue.
I am using RT-Linux (including Xenomai and Preempt-RT). According to chip manufacturers, A7's real-time nature will be better than A35, and then 32-bit will be better than 64-bit systems. So in the case of real -time system, ARM32 will be better than ARM64?

**Comment by n1ckfg** [comment]
My understanding is, since 64-bit OSes and apps use more RAM, on a very
RAM-constrained system like the Zero (512MB) you're hitting the SD card
swap.

Nick

**Comment by dimitre** [comment]
One question for anybody more in touch with RPI development: were we using GLFW for all architectures before?
